lib/rex/socket.rb

```diff
@@ -186,17 +186,12 @@ def self.getaddresses(hostname, accept_ipv6 = true)
       return [hostname]
     end
 
-    res = ::Socket.gethostbyname(hostname)
-    return [] if not res
+    res = ::Addrinfo.getaddrinfo(hostname, 0, ::Socket::AF_UNSPEC, ::Socket::SOCK_STREAM)
 
-    # Shift the first three elements out, leaving just the list of
-    # addresses
-    res.shift # name
-    res.shift # alias hostnames
-    res.shift # address_family
+    res.map! do |address_info|
+      address_info.ip_address
+    end
 
-    # Rubinius has a bug where gethostbyname returns dotted quads instead of
-    # NBO, but that's what we want anyway, so just short-circuit here.
```

Comment:
https://github.com/rapid7/rex-socket/pull/27#discussion_r577663928

Reply:
Further context to the above comment

The rubinus workaround can be deleted, however the filtering needs to be left in place. I believe it's probably something like this:

```
addr_info = ::Addrinfo.getaddrinfo(hostname, 0, ::Socket::AF_UNSPEC, ::Socket::SOCK_STREAM)
res.map! do |address_info|
    address_info.ip_address
end

unless accept_ipv6
    res.reject! { |ascii| ascii =~ MATCH_IPV6 }
end

res
```

If you wanted to add tests to verify the behavior is still the same, that works for me 🎉

Reply:
Got these tests added 👍